deployment/utility/sftp_copy.py

```diff
@@ -0,0 +1,95 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+from ..clusterObjectModel import cluster_object_model
+from ..k8sPaiLibrary.maintainlib import common
+import logging
+import logging.config
+
+
+class OpenPaiSftpCopy:
+
+    def __init__(self, filename, source, dest, cluster_config_path, filter):
+        self.filename = filename
+        self.source = source
+        self.dest = dest
+        objectModelFactoryHandler = cluster_object_model.cluster_object_model(configuration_path=cluster_config_path)
+        self.cluster_object_model = objectModelFactoryHandler.run()
+        self.filter_rule = filter
+        self.machine_list = {}
+
+        self.logger = logging.getLogger(__name__)
+
+
+    def construct_machine_list(self):
+        rule_list = []
+        self.logger.info("=============================================")
+        self.logger.info("================ Filter Rule ================")
+        self.logger.info("=============================================")
+        if self.filter_rule != None:
+            for rule in self.filter_rule:
+                kv = rule.split("=")
+                rule_list.append({"key":kv[0], "value":kv[1]})
+                self.logger.info("key = {0}, value = {1}".format(kv[0], kv[1]))
+        else:
+            self.logger.info("No filter rule.")
+        self.logger.info("\n")
+        self.logger.info("\n")
+
+        self.logger.info("=============================================")
+        self.logger.info("======= Machine List After filtered =========")
+        self.logger.info("=============================================")
+        for hostname in self.cluster_object_model['machine']["machine-list"]:
+            host = self.cluster_object_model['machine']["machine-list"][hostname]
+            flag = True
+            for rule in rule_list:
+                if rule["key"] not in host:
+                    flag = False
+                    break
+                if host[rule["key"]] != rule["value"]:
+                    flag = False
+                    break
+            if flag is True:
+                self.machine_list[hostname] = host
+                self.logger.info("Machine Host Name: {0},   Machine Ip Address: {1}".format(hostname, host["hostip"]))
+        self.logger.info("\n")
+        self.logger.info("\n")
+
+        count_input = 0
+        while True:
+            user_input = raw_input("Do you want to continue this operation? (Y/N) ")
+            if user_input == "N":
+                sys.exit(1)
+            elif user_input == "Y":
+                break
+            else:
+                print(" Please type Y or N.")
+            count_input = count_input + 1
+            if count_input == 3:
+                logger.warning("3 Times.........  Sorry,  we will force stopping your operation.")
+                sys.exit(1)
+
+    def run(self):
```

Comment:
We'd better choose some name more readable.
For example, it could be: uploade_xxx_ and so on.

Reply:
I don't think so. I think the unified entry point could make more sense.

Reply:
`run` is a too common name, which should be avoided.